deepchecks/vision/base.py

```diff
@@ -158,6 +158,33 @@ class SingleDatasetCheck(SingleDatasetBaseCheck):
 
     context_type = Context
 
+    def run_logic(self, context, dataset_type: str = 'train') -> CheckResult:
+        """Run check."""
+
+        self.initialize_run(context)
```

Comment:
isn't the initialize run should be called once from the suite?
what the purpose of having it here?